<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>
  <script>
    const getJSON = function(url) {

      const promise = new Promise((resolve, reject) => {
        const handler = function() {
          if (this.readyState !== 4) {
            return
          }
          if (this.status === 200) {
            resolve(this.response)
          }
          else {
            reject(new Error(this.statusText))
          }
        }

        const client = new XMLHttpRequest()
        client.open('GET', url)
        client.onreadystatechange = handler
        client.responseType = 'json'
        client.setRequestHeader('Accept', 'application/json')
        client.send()
      })

      return promise
    }

    // getJSON('./data.json').then((json) => {
    //   console.log(json)
    // }, (error) => {
    //   console.error('出错了', error)
    // })

    // 一般来说，不要在 then 方法里面定义 Reject 状态的回调函数，而是使用 catch 方法
    getJSON('./data.json').then((json) => {
      console.log(json)
    }).catch((error) => {
      console.error('出错了', error)
    })
  </script>
</body>
</html>